src/or/dirauth/dirvote.c

```diff
@@ -267,10 +267,13 @@ format_networkstatus_vote(crypto_pk_t *private_signing_key,
     else
       params = tor_strdup("");
     tor_assert(cert);
-    if (v3_ns->bw_file_headers)
+    if (v3_ns->bw_file_headers) {
       bw_file_headers = smartlist_join_strings(v3_ns->bw_file_headers, " ", 0,
```

Comment:
Please limit the length of the bw_file_headers string, not just the number of headers:
https://github.com/torproject/tor/pull/126#discussion_r198677005

(Limiting the number of headers to 50 header is not enough to limit the vote line to 1024 bytes, because each header can be 512 bytes long.)

Reply:
You could log a BUG() warning and use the empty string if the string is too long.
(It's very unlikely that the string will be too long.)
Remember to free the string before replacing it with "".

Reply:
> (Limiting the number of headers to 50 header is not enough to limit the vote line to 1024 bytes, because each header can be 512 bytes long.)

Is there already a constant limiting the size of the vote line?, i don't find something like that, maybe i should add it?.

Reply:
It would be:
```
/** Maximum size of a line in a vote. */
MAX_VOTE_LINE_LEN 1024
```
In ``dirvote.h``

Reply:
There are similar line length constants in diserv.h:
https://github.com/torproject/tor/blob/master/src/or/dirserv.h#L29
So I think this new constant should go in dirserv.h, even though it's only used in votes.

You said:
> /** Maximum size of a line in a vote. */
> MAX_VOTE_LINE_LEN 1024
But this constant does not limit all lines in the vote, just the bandwidth-file-headers line.

Please name the line limit constant after the kind of line, something like:
MAX_BANDWIDTH_HEADERS_LINE_LEN

Reply:
Or MAX_BW_FILE_HEADERS_LINE_LEN to match MAX_BW_FILE_HEADERS_LEN.

Reply:
Now that we have two similar constants, we should change MAX_BW_FILE_HEADERS_LEN to MAX_BW_FILE_HEADER_COUNT_IN_VOTE. Otherwise people might get confused.